offchain/danaswap-api/test/Api.purs

```diff
@@ -0,0 +1,198 @@
+        <> Lookups.mintingPolicy liquidityMP
+        <> Lookups.unspentOutputs configAdrUtxos
+    )
+    ( Constraints.mustMintCurrencyWithRedeemer -- Pool id token
+
+        poolIdMph
+        (Redeemer $ toData unit)
+        poolID
+        one
+        <> Constraints.mustMintCurrencyWithRedeemer -- Liquidity tokens
+
+          (mintingPolicyHash liquidityMP)
+          (Redeemer $ List [ toData poolID, Constr zero [] ])
+          wrongID
+          one
+        <> Constraints.mustReferenceOutput configUtxo
+        <> Constraints.mustPayToScript
+          (validatorHash poolAdrVal)
+          (Datum $ toData unit) -- TODO real pool datum
+          DatumInline
+          idNft
+    )
+  void $ waitForTx (scriptHashAddress $ validatorHash poolAdrVal) txid
+  pure poolID
+
+-- TODO rework this when the real open pool gets updated
+openPoolMultipleTokens :: Protocol -> Contract () PoolId
+openPoolMultipleTokens { poolAdrVal, liquidityMP, poolIdMP, configUtxo } = do
+  poolID <- liftContractM "failed to make token name" $ mkTokenName =<< hexToByteArray "aaaa"
+  wrongID <- liftContractM "failed to make token name" $ mkTokenName =<< hexToByteArray "aabb"
+  let poolIdMph = mintingPolicyHash poolIdMP
+  poolIdCs <- liftContractM "hash was bad hex string" $ mpsSymbol poolIdMph
+  --liquidityMPH <- liftContractM "failed to hash mp" $ mintingPolicyHash liq
+  let idNft = Value.singleton poolIdCs poolID one
+  configVal <- configAddressValidator
+  configAdrUtxos <- getUtxos (scriptHashAddress $ validatorHash configVal)
+  txid <- buildBalanceSignAndSubmitTx
+    ( Lookups.mintingPolicy poolIdMP
+        <> Lookups.mintingPolicy liquidityMP
+        <> Lookups.unspentOutputs configAdrUtxos
+    )
+    ( Constraints.mustMintCurrencyWithRedeemer -- Pool id token
+
+        poolIdMph
+        (Redeemer $ toData unit)
+        poolID
+        one
+        <> Constraints.mustMintCurrencyWithRedeemer -- Liquidity tokens
+
+          (mintingPolicyHash liquidityMP)
+          (Redeemer $ List [ toData poolID, Constr zero [] ]) -- This should be correct but should probably be a named constant
+          poolID
+          one
+        <> Constraints.mustMintCurrencyWithRedeemer -- Liquidity tokens
+
+          (mintingPolicyHash liquidityMP)
+          (Redeemer $ Constr one []) -- This should be correct but should probably be a named constant
+          wrongID
+          one
+        <> Constraints.mustReferenceOutput configUtxo
+        <> Constraints.mustPayToScript
+          (validatorHash poolAdrVal)
+          (Datum $ toData unit) -- TODO real pool datum
+          DatumInline
+          idNft
+    )
+  void $ waitForTx (scriptHashAddress $ validatorHash poolAdrVal) txid
+  pure poolID
+
+-- TODO rework with real depositLiquidity
+depositLiquidityWrongTokenRightRedeemer :: Protocol -> PoolId -> Contract () Unit
+depositLiquidityWrongTokenRightRedeemer protocol@{ poolAdrVal, liquidityMP, poolIdMP } poolID = do
```

Comment:
By rework, do you mean that these functions will be removed and the tests will just be testing functions that are used directly in the API?

Reply:
I mean when depositLiquidity gets a more complete implementation because the pool validator isn't a placeholder anymore than this will need to be updated too.